### PR TITLE
Couchbase provider

### PR DESCRIPTION
**Pull request by tellisnz**
Apologies for the scope of this PR - it's probably best to check out the commits individually, the last one is just formatting - I thought that before we start refactoring a bit more we could apply some consistent formatting so have added checkstyle.

I've also refactored the gradle builds a bit to start using sub projects to get rid of a bit of duplication.

Please give this a run inside the intranet before merging to make sure all ok.

### CODE REVIEW

File: core/src/test/java/org/mache/InMemoryCacheLoader.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package org.mache;
+
```
Comment (jbowkett) [inline core/src/test/java/org/mache/InMemoryCacheLoader.java:2]: This one was in my sights too....

File: couchbase-ns/src/test/java/CouchbaseCacheLoaderMockedTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        thenBucketCreated();
+        thenClusterManagerAndBucketCreated(mockedCluster);
+    }
+
+    @Test
+    public void shouldDropAndCreateBucket() throws Exception {
+        givenCacheLoaderWith(SchemaOptions.CREATEANDDROPSCHEMA);
+
+        when(mockedManager.hasBucket(anyString())).thenAnswer(new Answer<Boolean>() {
+            boolean returnValue = false;
```
Comment (jbowkett) [inline couchbase-ns/src/test/java/CouchbaseCacheLoaderMockedTest.java]: Consider extracting the answer into a named thing, to aid readability of the test

File: integration-example/build.gradle [hunk 3]
```diff
@@ -1,52 +1,28 @@
...
-sourceCompatibility = 1.7
-version = '1.0'
-jar {
-  manifest {
-    attributes 'Implementation-Title': 'Integration-Example',
-            'Implementation-Version': version
-  }
-}
-
 repositories {
```
Comment (jbowkett) [inline integration-example/build.gradle:1]: DRY me a riverrrr

File: integration-example/src/main/java/org/mache/GettingCacheClient.java [hunk 4]
```diff
@@ -4,85 +4,83 @@
...
+    public static void main(String... commandLine) {
+        final Args args = parseArgs(commandLine);
+        final int count = args.count;
+        switch (args.cacheType) {
+            case Cassandra:
+                doCassandra(count);
+                break;
+            case Mongo:
+                doMongo(count);
+                break;
```
Comment (jbowkett) [inline integration-example/src/main/java/org/mache/GettingCacheClient.java:29]: I'll add couch into here

File: build.gradle [hunk 5]
```diff
@@ -1,46 +1,41 @@
...
-jar {
-    manifest {
-        attributes 'Implementation-Title': 'ExCache',
-                   'Implementation-Version': version
-    }
-}
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'eclipse'
+    apply plugin: 'idea'
```
Comment (jbowkett) [inline build.gradle:4]: This is loads better

File: core/src/test/java/org/mache/InMemoryCacheLoader.java (hunk 1 above)
Comment (tellisnz) [inline core/src/test/java/org/mache/InMemoryCacheLoader.java:2]: I couldn't let it fly once I'd seen it

File: couchbase-ns/src/test/java/CouchbaseCacheLoaderMockedTest.java (hunk 2 above)
Comment (tellisnz) [inline couchbase-ns/src/test/java/CouchbaseCacheLoaderMockedTest.java]: Will do as soon as I hit the office

File: integration-example/src/main/java/org/mache/GettingCacheClient.java (hunk 4 above)
Comment (tellisnz) [inline integration-example/src/main/java/org/mache/GettingCacheClient.java:29]: Only if you've got time, I don't consider the couch stuff done yet. Want to get a good handle on blueprint and fix the couchbase images. I think it's a straightforward change to a kubernetes.json file but am creating my own blueprint env at home so i can touch things without breaking them

**Pull request merged**